src/components/Table/types.ts

```diff
@@ -0,0 +1,13 @@
+import { ReactNode } from "react"
+
+export interface WithID {
+  id: string
+}
+
+export interface ColumnConfig<Datum extends WithID> {
+  label: string | null
+  getValue?: (datum: Datum) => ReactNode
```

Comment:
I think there's an opportunity here to make `getValue` a bit more strict by forcing its return type to be `string | number | boolean` and the benefit is that these values are also sortable, which would let you get rid of that `isSortable` check elsewhere.

The drawback is that one wouldn't be able to shove any JSX into the table

Reply:
Great catch—and it gets at something I've been thinking about. I think we do need a selector function (`getValue`) that gets the primitive value—so yeah, let's tighten this up. I think an optional second function could be provided that would then be passed that value to be interpolated into some JSX structure. We need `getValue` for sorting, but we can't sort JSX... so let's separate them.

Reply:
Yeah, I think @luiz-chagas makes a great point. Enforcing the return value is a nice way to avoid the type checking. Then you can just have a separate render callback that also returns the value to be rendered within jsx.

Reply:
Thinking through this a bit more—it could be possible that you may want to configure a column that doesn't require any `Datum` information and only uses the `renderCell` (or whatever we call it) function to render something. In this case, if `getValue` was required, the user would still need to define it and return a `string | number | boolean` just to ignore it in `renderCell`.

I'm going to think through some options here.

Reply:
Maybe sortable can be used to discriminate whether getValue is required 😄

Reply:
Or, maybe rethink it entirely. Maybe instead of `getValue` it's `sortBy`

Reply:
> Or, maybe rethink it entirely. Maybe instead of `getValue` it's `sortBy`

@thispastwinter I think you're onto something there. I'm not sure we can type what I've been envisioning. It would be much easier to just pass the entire item into both functions. Thanks for helping me think outside the box I was stuck in! 🙏

Reply:
I think the nice thing about sortBy is the it removes the need for a Boolean.